### ISSUE TITLE
Packagists Version numbers not being updated

### ISSUE THREAD

**Issue by danmcadams** [badges/shields#2435]
Are you experiencing an issue with...
 - [X] [shields.io](https://shields.io/#/)
 - [ ] My own instance
 - [ ] [gh-badges NPM package](https://www.npmjs.com/package/gh-badges)

:beetle: **Description**
The version number on my packagist shield is not being updated, it is however the most recent version when you go to the page on packagist.  I checked the other issues here and they don't seem to line up with my issue. My versions are tagged exactly the same, and I have no version listed in composer.

Current version is v0.1.0 and my shield shows v0.0.1

The repository in question is here: https://github.com/The-Road-Bunch/command-menu

:link: **Link to the badge**
https://img.shields.io/packagist/v/theroadbunch/command-menu.svg

:bulb: **Possible Solution**

**Comment by paulmelnikow** [comment]
Hi! Thanks for the report.

Confirmed. It looks like there is some lag. The new version is not reflected in the endpoint we're hitting:

https://packagist.org/packages/theroadbunch/command-menu.json

Based on the doc here https://packagist.org/apidoc#get-package-data it's cached for 12 hours, which explains the issue.

Maybe the version badge could be modified to use the "always up to date" endpoint those docs mention, instead:

http://repo.packagist.org/p/theroadbunch/command-menu.json

**Comment by paulmelnikow** [comment]
@chris48s Do you know any reason (other than historical) that we might be using the dynamic endpoint for the version?

**Comment by chris48s** [comment]
No. Some of the packagist badges use `/p` and some use `/packages` but I don't think there is any solid reason for that (although note that the format of those 2 endpoints is _slightly_ different so we'd need to modify the parsing code a little bit to change).

**Comment by platan** [comment]
I think we can switch to "Composer metadata" (https://packagist.org/apidoc#get-package-data). I will be beneficial for Shields and for Packagist. About Composer metadata:
> This is the preferred way to access the data as it is always up to date, and dumped to static files so it is very efficient on our end.

**Comment by calebcartwright** [comment]
Sounds like a win-win

**Comment by PyvesB** [comment]
I believe we're using the composer API nowadays, so we no longer have this 12-hour cache issue. Will go ahead and close the issue.
